python/rpdk/java/templates/init/guided_aws/StubCreateHandler.java

```diff
@@ -29,157 +26,92 @@
+                            * While the handler contract states that the handler must always return a progress event,
+                            * you may throw any instance of BaseHandlerException, as the wrapper map it to a progress event.
+                            * Each BaseHandlerException maps to a specific error code, and you should map service exceptions as closely as possible
+                            * to more specific error codes
+                            */
+                            throw new CfnGeneralServiceException(ResourceModel.TYPE_NAME, e);
+                        }
+
+                        logger.log(String.format("%s successfully created.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
 
                     // STEP 2.3 [TODO: stabilize step is not necessarily required but typically involves describing the resource until it is in a certain status, though it can take many forms]
                     // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
-                    .stabilize(this::stabilizedOnCreate)
-                    .progress())
-
-            // STEP 3 [TODO: post create/stabilize update]
-            .then(progress ->
-                // If your resource is provisioned through multiple API calls, you will need to apply each subsequent update
-                // STEP 3.0 [initialize a proxy context]
-                proxy.initiate("{{ call_graph }}::post{{ operation }}", proxyClient, model, callbackContext)
-
-                    // STEP 3.1 [TODO: construct a body of a request]
-                    .translateToServiceRequest(Translator::translateToSecondUpdateRequest)
-
-                    // STEP 3.2 [TODO: make an api call]
-                    .makeServiceCall(this::postCreate)
+                    // If your resource requires some form of stabilization (e.g. service does not provide strong consistency), you will need to ensure that your code
+                    // accounts for any potential issues, so that a subsequent read/update requests will not cause any conflicts (e.g. NotFoundException/InvalidRequestException)
+                    .stabilize((awsRequest, awsResponse, client, model, context) -> {
+                        // TODO: put your stabilization code here
+
+                        final boolean stabilized = true;
+                        logger.log(String.format("%s [%s] has been stabilized.", ResourceModel.TYPE_NAME, model.getPrimaryIdentifier()));
+                        return stabilized;
+                    })
                     .progress()
                 )
 
-            // STEP 4 [TODO: describe call/chain to return the resource model]
+            // STEP 3 [TODO: describe call/chain to return the resource model]
             .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```

Comment:
Why is this read call here okay but not the pre-create read?

Reply:
read call at the end is to get a resource model with all the properties provisioned. pre check needs to be registered in the call chain as it gets reused. final step is not reused